assets/hdfs_append/index.js

```diff
@@ -21,6 +21,68 @@ function getClient(context, config, type) {
     return context.foundation.getConnection(clientConfig);
 }
 
+// Records the name of the offending file with a detected corrupt block in `appendErrors`
+function _recordFileError(name, errorLog) {
+    let newFilename = '';
+    // Get the original file name from the error message. If there have been no previous errors
+    // for the given file, this will result in a bogus filename that will trigger the second
+    // block of the following checks. Otherwise, the third block will be used to get a new name
+    const fileBase = name
+        .split('.')
+        .reverse()
+        .splice(1)
+        .reverse()
+        .join('.');
+    if (!errorLog.retry) {
+        // In this case, there has not been any hdfs append errors for a worker
+
+        newFilename = `${name}.0`;
+        errorLog[name] = newFilename;
+        // Ensures this block will not be executed again after the first error for a file
+        errorLog.retry = true;
+    } else if (!errorLog[fileBase]) {
+        // In this case, there has been at least one hdfs append error for a worker, but the
+        // worker is now writing to a new file
+
+        // Clean out the appendErrors object to avoid an accumulation of filenames
+        Object.keys(errorLog).forEach((key) => {
+            delete errorLog[key];
+        });
```

Comment:
Any real need for the loop instead of just `errorLog = []` to clear the array?

Reply:
I think the only reason why I have it like that is because I used `const` to define that object initially, but I can define it with `let` and simplify that part

Reply:
This is needed for circumventing a scoping constraint on the variable